spec.html

```diff
@@ -592,8 +593,11 @@ <h1>InnerModuleEvaluation ( _module_, _stack_, _index_ )</h1>
               1. <ins>Set _module_.[[PendingAsyncDependencies]] to _module_.[[PendingAsyncDependencies]] + 1.</ins>
               1. <ins>Append _module_ to _requiredModule_.[[AsyncParentModules]].</ins>
         1. <del>Perform ? _module_.ExecuteModule().</del>
-        1. <ins>If _module_.[[PendingAsyncDependencies]] &gt; 0, set _module_.[[AsyncEvaluating]] to *true*.</ins>
-        1. <ins>Otherwise, if _module_.[[Async]] is *true*, perform ! ExecuteAsyncModule(_module_).</ins>
+        1. <ins>If _module_.[[PendingAsyncDependencies]] &gt; 0 or _module_.[[Async]] is *true*, then</ins>
```

Comment:
We will end up wrapping non-async modules in promises, due to this line. Not sure yet how I feel about it.

Reply:
This is mostly an editorial refactoring. `[[AsyncEvaluating]]` was formerly set in `ExecuteAsyncModule`, we bring that set out of that function and set it here for both of the previous two conditional paths.

I only wanted to combine these two conditions into one because in both cases we want the same ordering note to apply, which otherwise would have needed to be repeated in both paths.

So the only semantic change really is that note, the rest is just editorial refactoring to permit a single note.

Reply:
Strictly speaking, the order of [[AsyncEvaluating]] does only matter in the [[PendingAsyncDependencies]] > 0 case so that we could just have the note in the first condition and leave the logic as before with the diff just being a note on the first line of this diff.

Reply:
OK, good! I most definitely was surprised because it didn't fit my mental model.